src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -181,22 +182,42 @@ def _normalize_properties(user, subscriptions, is_service_principal):
 
     @staticmethod
     def _build_tenant_level_accounts(tenants):
-        from azure.cli.core.profiles import get_sdk, ResourceType
-        SubscriptionType = get_sdk(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS,
-                                   'Subscription', mod='models')
-        StateType = get_sdk(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS,
-                            'SubscriptionState', mod='models')
         result = []
         for t in tenants:
-            s = SubscriptionType()
+            s = Profile._new_account()
             s.id = '/subscriptions/' + t
             s.subscription = t
             s.tenant_id = t
             s.display_name = 'N/A(tenant level account)'
-            s.state = StateType.enabled
             result.append(s)
         return result
 
+    @staticmethod
+    def _new_account():
+        from azure.cli.core.profiles import get_sdk, ResourceType
+        SubscriptionType = get_sdk(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS,
+                                   'Subscription', mod='models')
+        StateType = get_sdk(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS,
+                            'SubscriptionState', mod='models')
+        s = SubscriptionType()
+        s.state = StateType.enabled
+        return s
+
+    def init_if_in_msi_env(self, user):
+        _, subscription_id = MsiCreds.split_user_info(user)
+        if subscription_id is None:
+            return None
+        logger.debug('MSI: environment was detected. Now trying to initialize a local account...')
```

Comment:
Perhaps this should be an info statement?  For example, if I ran with --verbose, I would want to see this.